### PR TITLE
33 3 funktion filtrera film kort

### LINKED ISSUE #33: [3] Funktion: Filtrera film kort
## Details
Create a function that will take the full array of movies and return a filtered version, depending on a parameter.

Visningstider: Return movies that have a showtime in the coming 2 days.
På bio just nu: Return max 21 random.
Karusell: Return 4 random.
Kommande premiärer: Movies with a premier date spanning from today onwards.
Klassiker: Movies released up to 1990.
Salong x: Movies shown in salon x.

Example (pseudo code)
```
function filterMovies(array, parameter) {
  let results = [];
  switch (parameter)
    case "1990": 
        results = array.filter(movie => movie.premiere < 1990)
  break;
  return results;
}
```

## User story
As a developer, I need a function to be able to get the movies I need to show at this particular page.

## Definition of done
You should be able to send in an array of movies and get the results depending on what parameter you send in.

## Tests
If you send an array of movies with the parameter 1990, the result you get back should be movies released before 1990.

### PR DESCRIPTION
**Pull request by saeraa**
Added file: `js/filterMovies.js`

To test the functions parameters:
```
//console.log(await filterMovies([], "visningstider"));
//console.log(await filterMovies([], "karusell"));
//console.log(await filterMovies([], "salongalcazar"));
//console.log(await filterMovies([], "salongbardeco"));
//console.log(await filterMovies([], "salongcamera"));
//console.log(await filterMovies([], "kommande"));
//console.log(await filterMovies([], "pabionu"));
//console.log(await filterMovies([], "klassiker"));
```

Some formatting things in placeholder.html that you can ignore, I had default formatter before instead of agreed-upon Prettier so it got messy again 😣

### CODE REVIEW

File: js/filterMovies.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+			now.setHours(0, 0, 0, 0);
+
+			tomorrow = new Date();
+			tomorrow.setDate(now.getDate() + 1);
+			tomorrow.setHours(23, 59, 59, 59);
+
+			movieTimes.forEach((salon) => {
+				salon.dates.forEach((date) => {
+					const showtime = new Date(date.date).getTime();
+					if (now.getTime() > showtime && showtime < tomorrow.getTime()) {
```
Comment (Arash-SJPI22) [inline js/filterMovies.js]: It should be the other way arround  showtime > now.getTime()  otherwise it pass dates that are before now :D

**Comment by Arash-SJPI22** [comment]
It looks good, just need to change the one bug I found.
I would recommend you to add code comments to make it easier to grasp the code quicker :D
Personally I think it is easier to work with dates in a human readable format and it makes bug hunting easier without the extra effort to convert all the dates.

Also maybe use English variables for the switch case for consistency :D

**Review by Arash-SJPI22 — CHANGES_REQUESTED** [review]
Just need to fix 1 bug :+1:

File: js/filterMovies.js (hunk 1 above)
Comment (saeraa) [reply js/filterMovies.js]: 👀 Noooooooo

Fixed

**Review by Arash-SJPI22 — APPROVED** [review]
Superb!
Nice with variable name changes too :)

**Pull request merged**